drivers/timer/nrf_grtc_timer.c

```diff
@@ -426,8 +426,12 @@ int z_nrf_grtc_wakeup_prepare(uint64_t wake_time_us)
 		nrfy_grtc_timeout_get(NRF_GRTC) * CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / 32768 +
 		MAX_CC_LATCH_WAIT_TIME_US;
 	k_busy_wait(wait_time);
-#if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(lfxo)) && NRF_GRTC_HAS_CLKSEL
+#if NRF_GRTC_HAS_CLKSEL
+#if defined(CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC)
+	nrfx_grtc_clock_source_set(NRF_GRTC_CLKSEL_LFLPRC);
+#elif DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(lfxo))
 	nrfx_grtc_clock_source_set(NRF_GRTC_CLKSEL_LFXO);
```

Comment:
should there be
```
#else
#error Lack of LFCLK source

Reply:
`CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC=n` doesn't mean that there is no RC source (it is internal). In such case the GRTC will be sourced from LFCLK oscillator control - the configuration remains unchanged since the  GRTC was initialized: https://github.com/zephyrproject-rtos/hal_nordic/blob/master/nrfx/haly/nrfy_grtc.h#L210

Reply:
In such case the configuration introduced by the GRTC init will remain unchanged (https://github.com/zephyrproject-rtos/hal_nordic/blob/master/nrfx/haly/nrfy_grtc.h#L210) - LFCLK oscillator control as a source